### PR TITLE
Enhance viewpoint filter

### PR DESCRIPTION
**Pull request by jrmi**

### CODE REVIEW

File: src/utils/date.js [hunk 1]
```diff
@@ -11,4 +11,8 @@ export const localErrorMessages = {
 
 export const locale = i18n.language;
 
+/* ISO 8601 date format compatible with django filter */
+export const ISODateFormat = date =>
+  `${date.getFullYear() - 1}-${`${date.getMonth() + 1}`.padStart(2, '0')}-${date.getDate()}`;
```
Comment (dtrucs) [inline src/utils/date.js:16]: Maybe it's shorter in the following way
```js
export const ISODateFormat = date => date.toISOString().split('T')[0]
```

**Review by dtrucs — APPROVED** [review]

**Review by CFarcy — APPROVED** [review]

**Review by vllorca — DISMISSED** [review]

File: src/modules/RA/Viewpoint/views/ListFilters.js [hunk 2]
```diff
@@ -39,41 +46,72 @@ export const ListFilters = ({ terraOppSearchableProperties, ...props }) => {
...
+    const more2Years = ISODateFormat(new Date(new Date().setFullYear(currentYear - 2)));
+    const more3Years = ISODateFormat(new Date(new Date().setFullYear(currentYear - 3)));
+    const more5Years = ISODateFormat(new Date(new Date().setFullYear(currentYear - 5)));
+
+    return [
+      { id: more1Year, name: 'resources.viewpoint.filters.more-1-year' },
+      { id: more2Years, name: 'resources.viewpoint.filters.more-2-year' },
+      { id: more3Years, name: 'resources.viewpoint.filters.more-3-year' },
+      { id: more5Years, name: 'resources.viewpoint.filters.more-5-year' },
+    ];
```
[the remark this answers is not in the document]
Comment (jrmi) [reply src/modules/RA/Viewpoint/views/ListFilters.js]: I dried the isodate part.

File: src/utils/date.js (hunk 1 above)
Comment (jrmi) [reply src/utils/date.js]: I don't like tinkering with strings but I don't have any good argument to oppose, so I did it.

**Review by dtrucs — DISMISSED** [review]

File: public/locales/fr/translation.json [hunk 3]
```diff
@@ -906,6 +906,12 @@
         "repeat": "Rephotographie",
         "landscape": "Paysage",
         "pictures": "Photographies"
+      },
+      "filters": {
+        "more-1-year": "Plus d'un an",
+        "more-2-year": "Plus de 2 ans",
+        "more-3-year": "Plus de 3 ans",
+        "more-5-year": "Plus de 5 ans"
```
Comment (dtrucs) [inline public/locales/fr/translation.json:914]: No "en" translations?

**Pull request merged**